client/my-sites/stats/activity-log/index.jsx

```diff
@@ -234,17 +234,17 @@ class ActivityLog extends Component {
 		const logsGroupedByDay = map(
 			groupBy(
 				logsForMonth,
-				log => applySiteOffset( moment.utc( log.ts_utc ) ).format( 'LL' )
+				log => applySiteOffset( moment.utc( log.ts_utc ) ).startOf( 'day' ).valueOf()
 			),
-			( daily_logs, day ) => (
+			( daily_logs, tsDayStart ) => (
 				<ActivityLogDay
 					allowRestore={ !! isPressable }
 					isRewindActive={ isRewindActive }
-					key={ day }
+					key={ tsDayStart }
 					logs={ daily_logs }
 					requestRestore={ this.handleRequestRestore }
 					siteId={ siteId }
-					day={ day }
+					tsDayStart={ +tsDayStart }
```

Comment:
Yeah, I now think you are right, this PR is doing the right things. My unease comes from having a prop `tsDayStart` which is the start of the local day, not the start of the utc day.

> The start of a day should be localized to site time, as that does change.

Mm, this is true. On the other hand, for a prop in unix time format, I might assume it means the start of the utc day.

How about keeping this PR as as, but changing the prop to `tsEndOfLocalDay`, which makes it slightly more explicit and avoids the need to do `endOf()` down in the day component? This assumes `endOf()` gives a stamp which is still inside the day.